### PR TITLE
Prepare covidcast-R for release 0.5.0

### PR DESCRIPTION
**Pull request by krivard**
Includes:
 * Change to the multi-signal vignette: the one-state modification to fit into anonymous request limits had pushed all the subsequent operations to an empty data frame. I dropped the filters and gave you back Texas 🤠 
 * Version bump
 * Date bump
 * Documentation build - [preview](http://rinkitink.ml.cmu.edu/covidcast/covidcastR/index.html)

I cleared out all the old docs before building the documentation site, so there are some file removals in this PR for files which are no longer produced by pkgdown. I glanced through them and the only real suspicious one is reference/covidcast.html, but its last modification date in git is from 2 years ago, and I couldn't find anywhere in the current or candidate documentation site that actually links to it; I think all that content got moved to index ages ago.

### CODE REVIEW

**Review by capnrefsmmat — APPROVED** [review]
I've made some improvements here, and it now passes a check on the R win-builder and on my machine. Would appreciate a double-check of the changes to `covidcast.R`, the README, and NEWS

**Review by dshemetov — APPROVED** [review]
Gave Alex's recent changes a look. Seem fine to me 👍

**Comment by capnrefsmmat** [comment]
Great, I'll go ahead and proceed with CRAN submission.

File: R-packages/covidcast/NEWS.md [hunk 1]
```diff
@@ -1,9 +1,12 @@
 # covidcast 0.5.0
 
-## Major changes
+- The package now supports supplying API keys with requests to the COVIDcast
+  Epidata API. Beginning June 28, 2023, the Epidata API server will reject
```
Comment (krivard) [inline R-packages/covidcast/NEWS.md]:
```suggestion
  Epidata API. Beginning June 21, 2023, the Epidata API server will reject
```

**Comment by capnrefsmmat** [comment]
Okay, updated the form link and set it to extract error messages on 429 responses. If that's all, I can submit to CRAN soon.

**Comment by capnrefsmmat** [comment]
Just submitted 0.5.0 to CRAN. Hopefully it goes smoothly this time.

**Comment by capnrefsmmat** [comment]
Aaand approved:

> Dear maintainer,
>
> thanks, package covidcast_0.5.0.tar.gz is on its way to CRAN.

I'll merge this once the package appears on CRAN, so the documentation is updated accordingly.

**Pull request merged**